File: anul2/semestrul1/map/lab2/sort/SortingFactory.java
package sort;

public class SortingFactory implements SortingInterface {
    private static SortingFactory instance = null;

    private SortingFactory() {}

    public static SortingFactory getInstance() {
        if (instance == null) {
            instance = new SortingFactory();
        }
        return instance;
    }

    @Override
    public AbstractSorter makeSorter(SortingStrategy strategy) {
        return switch (strategy){
            case BubbleSort -> new BubbleSort();
            case MergeSort -> new MergeSort();
            default -> throw new IllegalArgumentException("Unsupported SortingStrategy");
        };
    }
}

File: Third Semester - Advanced Programming Methods/Lab10/sort/SortingFactory.cs
namespace lab10.sort;

public class SortingFactory: SortingInterface
{
    private static SortingFactory _instance = null;
    
    private SortingFactory() { }

    public static SortingFactory Instance => _instance ??= new SortingFactory();

    public AbstractSorter makeSorter(SortingStrategy strategy)
    {
        switch (strategy)
        {
            case SortingStrategy.BubbleSort: return new BubbleSort();
            case SortingStrategy.MergeSort: return new MergeSort();
            default: return null;
        }
    }
}